### PR TITLE
use process lock when loading dependency lock file

### PR DESCRIPTION
**Pull request by OsmondJiang**
change docset to async create, for loading dependency lock before building

### CODE REVIEW

File: src/docfx/docset/Docset.cs [hunk 1]
```diff
@@ -134,21 +172,20 @@ private Docset(Report report, string docsetPath, string locale, Config config, C
...
             DependencyLock = dependencyLock;
 
-            var configErrors = new List<Error>();
-            (configErrors, DependencyDocsets) = LoadDependencies(Config);
             ResolveAlias = LoadResolveAlias(Config);
             Repository = repository ?? Repository.Create(DocsetPath, branch: null);
 
             // pass on the command line options to its children
             _buildScope = new Lazy<HashSet<Document>>(() => CreateBuildScope(Redirections.Files));
+            _buildScope = new Lazy<HashSet<Document>>(() => CreateBuildScope(Redirections.Files));
```
Comment (yufeih) [inline src/docfx/docset/Docset.cs]: dup

File: test/docfx.Test/build/TocTest.cs [hunk 2]
```diff
@@ -9,13 +9,13 @@ namespace Microsoft.Docs.Build
...
     {
-        private static readonly Docset s_docset = new Docset(
+        private static readonly Docset s_docset = Docset.Create(
             new Report(),
             Directory.GetCurrentDirectory(),
             "en-us",
             JsonUtility.Deserialize<Config>("{'output': { 'json': true } }".Replace('\'', '\"')),
             new CommandLineOptions(),
-            null);
+            null).Result;
```
Comment (yufeih) [inline test/docfx.Test/build/TocTest.cs]: this test always needs update when we do docset refactor 😢

File: src/docfx/restore/lock/DependencyLock.cs [hunk 3]
```diff
@@ -46,10 +46,10 @@ public static DependencyLock Load(string docset, string dependencyLockPath)
             var (_, restoredLockFile) = RestoreMap.GetFileRestorePath(docset, dependencyLockPath);
 
             // todo: add process lock
```
Comment (yufeih) [inline src/docfx/restore/lock/DependencyLock.cs]: remove TODO

File: test/docfx.Test/build/TocTest.cs (hunk 2 above)
Comment (OsmondJiang) [reply test/docfx.Test/build/TocTest.cs]: :(

**Pull request merged**